src/hooks/useAccommodationsAndOffers.tsx/helpers.ts

```diff
@@ -157,8 +153,39 @@ export const accommodationEventTransform =
         focusedEventCoordinates[1]
       );
 
+      // return eventInfo as an array of distances with focusedEventInfo at the top if it exists
       const durationInMinutes = (distance / 5) * 60; // we are assuming 5km/hr walking distance in minutes
-      eventInfo = { distance, eventName: focusedEvent, durationInMinutes };
+      eventInfo.push({ distance, eventName: focusedEvent, durationInMinutes });
+    }
+
+    if (currentEventsWithinRadius && currentEventsWithinRadius.length) {
+      const infos: EventInfo[] = [];
+
+      for (let idx = 0; idx < currentEventsWithinRadius.length; idx++) {
+        const event = currentEventsWithinRadius[idx];
+
+        const eventCoordinates = event?.latlon && [event.latlon[0], event.latlon[1]];
+
+        if (eventCoordinates) {
+          const distance = crowDistance(
+            accommodation.location.coordinates[1],
+            accommodation.location.coordinates[0],
+            eventCoordinates[0],
+            eventCoordinates[1]
+          );
+
+          // return eventInfo as an array of distances with focusedEventInfo at the top if it exists
+
+          const durationInMinutes = (distance / 5) * 60; // we are assuming 5km/hr walking distance in minutes
+          infos.push({ distance, eventName: event.name, durationInMinutes });
+        }
+      }
+
+      // sort by distance with nearest events first
+      infos.sort((a, b) => a.distance - b.distance);
+
+      // update eventInfo array with focusedEvent at top and other events in ascending distance
+      eventInfo.push(...infos);
```

Comment:
should `eventInfo` and `info` not be a `let` ?

Reply:
They are initialised as empty arrays and are not being reassigned.
I considered using a let for`eventInfo`, but that'll require an extra step checking if the variable is an array or not and deciding when to initialise etc. Bottom line is it will get a bit messy